File: Lakshay Garg/Leetcode/Sorting/Merge Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& arr) {
        sort(arr.begin(),arr.end());
        vector<vector<int>>v;
        v.push_back(arr[0]);
        int i=0,j=0,n=arr.size();
        while(j<n){
            if(v[i][1]>=arr[j][0]){
                v[i][1]=max(v[i][1],arr[j][1]);
                j++;
            }
            else{
                v.push_back(arr[j]);
                i++;
                j++;
            }
        }
        return v;
    }
};

File: Anya_Seth/Sorting/Merge Intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (interval1, interval2) -> interval1[0] - interval2[0]);
        int end = intervals[0][1];
        List<int[]> answer = new ArrayList<>();
        for(int j = 0, i = 1 ; i <= intervals.length ; i++){
            if(i == intervals.length || intervals[i][0] > end){
                answer.add(new int[] {intervals[j][0], end});
                j = i;
            }
            if(i != intervals.length){
                end = Math.max(end, intervals[i][1]);
            }
        }
       return answer.toArray(new int[answer.size()][]);
    }
}
